kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/OtelMetricsService.java

```diff
@@ -0,0 +1,123 @@
+import io.opentelemetry.sdk.resources.Resource;
+import java.io.Closeable;
+import java.time.Duration;
+import org.apache.kafka.streams.StreamsConfig;
+
+public class OtelMetricsService implements Closeable {
+
+  private final JmxMetricInsight metricInsight;
+  private final Runnable onClose;
+
+  public static OtelMetricsService create(
+      final StreamsConfig streamsConfig,
+      final ResponsiveConfig config
+  ) {
+    final var exporter = OtlpGrpcMetricExporter
+        .builder()
+        .addHeader("api-key", config.getString(ResponsiveConfig.METRICS_API_KEY_CONFIG))
+        .addHeader("secret", config.getString(ResponsiveConfig.METRICS_API_KEY_CONFIG))
+        .setEndpoint(config.getString(ResponsiveConfig.CONTROLLER_ENDPOINT_CONFIG))
+        .build();
+
+    final var metricReader = PeriodicMetricReader
+        .builder(exporter)
+        .setInterval(Duration.ofSeconds(10))
+        .build();
+
+    final var appId = ConfigUtils.responsiveAppId(streamsConfig, config);
+    final var attributes = Attributes
+        .builder()
+        .put("service.name", appId + "-otel")
+        .put("responsiveApplicationId", appId)
+        .build();
+
+    final var meterProvider = SdkMeterProvider
+        .builder()
+        .setResource(Resource.create(attributes))
+        .registerMetricReader(metricReader)
+        .build();
+
+    final OpenTelemetrySdk otel = OpenTelemetrySdk
+        .builder()
+        .setMeterProvider(meterProvider)
+        .setPropagators(ContextPropagators.create(TextMapPropagator.composite(
+            W3CTraceContextPropagator.getInstance(),
+            W3CBaggagePropagator.getInstance()
+        )))
+        .build();
+
+    return new OtelMetricsService(otel, otel::close);
+  }
+
+  public static OtelMetricsService noop() {
+    return new OtelMetricsService(OpenTelemetry.noop(), () -> {});
+  }
+
+  private OtelMetricsService(final OpenTelemetry otel, final Runnable onClose) {
+    this.onClose = onClose;
+    this.metricInsight = JmxMetricInsight.createService(otel, 0);
+  }
+
+  public void start() {
+    this.metricInsight.start(buildMetricConfiguration());
+  }
+
+  @Override
+  public void close() {
+    onClose.run();
+  }
+
+  private static void buildFromUserRules(MetricConfiguration conf) {
+    RuleParser parserInstance = RuleParser.get();
+    final ClassLoader loader = OtelMetricsService.class.getClassLoader();
+
+    // TODO(agavra): instead of including otel-jmx.config.yaml as a resource we should
+    // fetch it from the Responsive controller on start-up
+    try (final var is = loader.getResourceAsStream("otel-jmx.config.yaml")) {
+      parserInstance.addMetricDefsTo(conf, is, "responsive builtin config");
+    } catch (Exception e) {
+      // yaml parsing errors are caught and logged inside addMetricDefsTo
+      // only file access related exceptions are expected here
+      JmxMetricInsight.getLogger().warning(e.toString());
```

Comment:
I think this should be a fatal error that we should throw the exception up for. Now that we're getting the config from a resource, this indicates something is seriously wrong with the jar we're shipping.